### PR TITLE
Preprocessing config selection

### PR DESCRIPTION
**Pull request by maximzubkov**

### CODE REVIEW

File: data_preprocessing/preprocess_code2seq_data.py [hunk 1]
```diff
@@ -154,6 +164,7 @@ def preprocess(config: PreprocessingConfig, is_vocab_collected: bool, n_jobs: in
     arg_parser.add_argument("data", type=str)
     arg_parser.add_argument("--collect-vocabulary", action="store_true")
     arg_parser.add_argument("--n-jobs", type=int, default=None)
+    arg_parser.add_argument("--problem", type=str, default="code2seq")
```
Comment (SpirinEgor) [inline data_preprocessing/preprocess_code2seq_data.py:167]: Same comment as for `train.py`. Set as a positional argument with choices.

File: data_preprocessing/preprocess_code2seq_data.py [hunk 2]
```diff
@@ -137,8 +146,11 @@ def convert_holdout(holdout_name: str, vocab: Vocabulary, config: PreprocessingC
         _ = [_ for _ in tqdm(results, total=n_buffers)]
 
 
-def preprocess(config: PreprocessingConfig, is_vocab_collected: bool, n_jobs: int):
+def preprocess(problem: str, data: str, is_vocab_collected: bool, n_jobs: int):
     # Collect vocabulary from train holdout if needed
+    config_function = _config_switcher[problem]
```
Comment (SpirinEgor) [inline data_preprocessing/preprocess_code2seq_data.py:151]: You should check the dict for the key and raise an error if the user passed an unknown problem.
```
if problem not in _config_switcher:
    raise
...
```

Comment (maximzubkov) [reply data_preprocessing/preprocess_code2seq_data.py:151]: But as far as I understood it is already checked by `argparse` https://docs.python.org/3/library/argparse.html#choices

File: data_preprocessing/preprocess_code2seq_data.py [hunk 3]
```diff
@@ -137,8 +146,14 @@ def convert_holdout(holdout_name: str, vocab: Vocabulary, config: PreprocessingC
         _ = [_ for _ in tqdm(results, total=n_buffers)]
 
 
-def preprocess(config: PreprocessingConfig, is_vocab_collected: bool, n_jobs: int):
+def preprocess(problem: str, data: str, is_vocab_collected: bool, n_jobs: int):
     # Collect vocabulary from train holdout if needed
+    if problem in _config_switcher:
+        config_function = _config_switcher[problem]
+    else:
+        raise ValueError(f"Unknown problem ({problem}) passed")
```
Comment (SpirinEgor) [inline data_preprocessing/preprocess_code2seq_data.py:154]:
```
if problem not in _config_switcher:
    raise ValueError(...)
config_function = _config_switcher[problem]
```
It takes less space and easy to read

**Pull request merged**